CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc

```diff
@@ -564,6 +564,28 @@ HcalQIECoder HcalDbHardcode::makeQIECoder (HcalGenericDetId fId) {
   return result;
 }
 
+HcalQIEType HcalDbHardcode::makeQIEType (HcalGenericDetId fId, bool testHFQIE10) {
+  int qieType = HcalQIEType::QIE8; //default
+  if (testHFQIE10){ //2016 test
+    if(fId.genericSubdet() == HcalGenericDetId::HcalGenForward && fId.isHcalDetId()) {
+        HcalDetId hid(fId);
+        if(hid.depth()>=3) qieType = HcalQIEType::QIE10;
+    }
```

Comment:
@franzoni - In the 2016 HF, the new QIE10 readouts/cells are marked as depths 3/4 (corresponding to the existing HF depths 1/2). This is allowed and supported by the new topology code. Slide 4 in the talk you reference is "till 2015", i.e. it does not address the 2016+ detector. It is important to note that depths 3/4 will never survive to the RECO stage, so HF at the RecHit level will still only have depths 1/2. In 2016, the QIE10 cells will just be ignored for RECO; in 2017, we will have full dual anode readout, and then depths 1 and 3 (2 and 4) will be combined intelligently to make a single RecHit.

HcalDbHardcode is only used for testing purposes. It will not be used for the "old" scenarios. You can further see that the default is set to QIE8:
```
int qieType = HcalQIEType::QIE8; //default
```
and the `testHFQIE10` flag is only enabled with the 2016 era. There should not be any issues maintaining the old scenarios.